plugins/modules/system/launchd.py

```diff
@@ -163,7 +163,7 @@ def __find_service_plist(self, service_name):
 
         for path in launchd_paths:
             try:
-                files = os.listdir(os.path.expanduser(path))
+                files = os.listdir(path)
```

Comment:
Can someone explain what actually the difference is? I already questioned this back then when this problem was reported and wondering why there is a different behaviour depending when `expanduser` is called.

This simple script works as I would expected, namely no difference:

```
import os

path = '~/Library/LaunchAgents'
files1 = os.listdir(os.path.expanduser(path))

path = os.path.expanduser('~/Library/LaunchAgents')
files2 = os.listdir(path)

assert files1 == files2
```

Reply:
Sorry, my mistake. I overlooked this line:

```
return os.path.join(path, filename)
```

which uses the non-expanded path :doh:

Reply:
Hi Martin, the problem happened later on line:
```
     return os.path.join(path, filename)
```
The `path` variable was taken from the array (not expanded), so `open()` got a literal `~` in the filename.

BTW, thanks for your great job on this module!

Reply:
Thanks @tomaszn, I indeed missed that line. Would you change a bit your fix and to the expand within the for-loop and add as well the `~/Library/LaunchDaemons` to the array (or we can add this additional directory in a separate PR).

I think it would as well make sense to write an integration test for a plist file within the home directory.

Reply:
@martinm82: sure, will add that path too.

But I have objections against calling `expanduser` inside the for loop:
1. There is no purpose of calling it on paths that contain no user variables.
2. We had the call inside the loop and a bug happened. I think it's more readable and safer to expand the paths as soon as possible.

Reply:
Adding that path requires another entry in the changelog fragment (section `minor_changes`).

About expanduser: it is a rather cheap call, so doing it in the loop is OK. If it is done once at the beginning of the loop and its result stored as `path` again, it's also pretty safe.